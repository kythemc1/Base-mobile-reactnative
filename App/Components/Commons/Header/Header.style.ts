import {colors} from 'Configs/UI/Colors';
import sizes from 'Configs/UI/Sizes';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
    container: {backgroundColor: colors.color_background},
    mContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: colors.color_background,
        borderBottomRightRadius: sizes._20sdp,
        borderBottomLeftRadius: sizes._20sdp
    },
    mBody: {
        width: '90%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    mTitle: {
        color: colors.color_text_grey1,
        fontSize: sizes._18sdp,
        fontWeight: 'bold',
        textAlign: 'left',
        marginRight: sizes._115sdp,
    },
    mTitleContainer: {
        alignItems: 'flex-start',
    },
    imgLogo: {
        height: sizes._40sdp,
        width: sizes._100sdp
    },
    avatar: {
        width: sizes._34sdp,
        height: sizes._34sdp,
        borderRadius: sizes._20sdp,
        marginRight: sizes._15sdp
    },

    viewLogo: {
        marginLeft: sizes._screenWidth * 0.05,
        height: sizes._40sdp,
        width: sizes._100sdp
    },


    bell: {
        width: sizes._24sdp,
        height: sizes._24sdp,

    },
    viewAvatar: {
        width: sizes._screenWidth * 0.2,
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    viewNoti: {
        height: sizes._12sdp,
        width: sizes._12sdp,
        backgroundColor: colors.color_primary2,
        borderRadius: sizes._10sdp,
        position: 'absolute',
        left: sizes._10sdp,
        top: sizes._4sdp,
    },
    textInNoti: {
        position: 'absolute',
        color: colors.color_white,
        fontSize: sizes._8sdp,
        alignSelf: 'center'
    },
    mBadge: {
        position: 'absolute',
        right: sizes._65sdp,
        top: sizes._15sdp
    }
});
