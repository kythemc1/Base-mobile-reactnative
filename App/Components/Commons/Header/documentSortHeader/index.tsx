import {Image, Text, TouchableOpacity, View} from 'react-native';
import icons from 'Configs/Constants/icons';
import React from 'react';
import {styles} from 'Components/Commons/Header/documentSortHeader/styles';
import { useSelector } from 'react-redux';
import { RootState } from 'Store/reduxProvider';

interface sortInterface {
    onPressView?: () => void;
    statusIcon?: boolean;
    onPressSort?: () => void;
    statusSortIcon?: boolean;
    title?:string
}

export const Sort = (props: sortInterface) => {
    return (
        <View
            style={styles.mContainer}>
            <TouchableOpacity
                onPress={props.onPressSort}
                style={styles.mSort}>
                <Image
                    source={props.statusSortIcon ? icons.ic_sort_asc : icons.ic_sort_desc}
                    style={styles.icon}
                    resizeMode={'contain'}
                />
                <Text style={styles.mTitle} >{props.title}</Text>
                <Image
                    source={icons.ic_arrow_down}
                    style={styles.icon}
                    resizeMode={'contain'}
                />
            </TouchableOpacity>
            <TouchableOpacity onPress={props.onPressView}>
                <Image
                    source={props.statusIcon ? icons.ic_elements : icons.ic_view_list}
                    style={styles.imgView}
                    resizeMode={'contain'}
                />
            </TouchableOpacity>
        </View>
    );
};