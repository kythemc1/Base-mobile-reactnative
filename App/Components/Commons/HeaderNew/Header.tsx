import React from 'react';
import {
    View,
    Image, TouchableOpacity,
} from 'react-native';
import {useSelector} from 'react-redux';
import {RootState} from '../../../Store/reduxProvider';

import icons from 'Configs/Constants/icons';
import {Appbar, Avatar, Badge} from 'react-native-paper';
import {styles} from 'Components/Commons/HeaderNew/HeaderStyles';
import images from 'Configs/Constants/images';

interface headerInterface {
    onPress: () => void;
    onShowProfile: () => void;
}

export default function Header(props: headerInterface) {
    const name = useSelector((state: RootState) => state.auth.user?.username);
    return (
        <Appbar.Header mode={'center-aligned'} style={{backgroundColor: 'transparent'}}>
            <View style={styles.viewLogo}>
                <Image
                    style={styles.bellringing}
                    source={images.img_logo_sidoc}
                />
            </View>
            <Appbar.Content title={''}/>
            <Appbar.Action icon={icons.ic_noti} onPress={props.onPress}/>
            <Badge style={{position: 'absolute', right: 65, top: 15}} size={15}>99+</Badge>
            <TouchableOpacity style={{width: 32, height: 32, borderRadius: 20, marginRight: 15}}
                onPress={props.onShowProfile}>
                <Avatar.Image source={{uri: 'https://i.pravatar.cc/300'}} size={32}/>
            </TouchableOpacity>

        </Appbar.Header>
    );
}

