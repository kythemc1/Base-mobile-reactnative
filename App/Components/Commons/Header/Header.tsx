import React, {useState} from 'react';
import icons from 'Configs/Constants/icons';
import {Appbar, Avatar, Badge} from 'react-native-paper';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import sizes from 'Configs/UI/Sizes';
import {colors} from 'Configs/UI/Colors';
import {useTranslation} from 'react-i18next';
import {styles} from './Header.style';
import images from 'Configs/Constants/images';
import Sizes from 'Configs/UI/Sizes';
import {SideMenu} from 'Navigations/SideMenu/SideMenu';
import {SideSetting} from 'Navigations/SideSetting/SideSetting';
import {AVATAR_LINK} from '@env';
import {useSelector} from 'react-redux';
import {RootState} from 'Store/reduxProvider';

interface headerInterface {
  onPress?: () => void;
  statusSelect?: boolean;
  onPressSearch?: () => void;
  typeHeader?: 'normal' | 'back_header' | 'search_header';
  onBackPress?: () => void;
  titleHeader?: any;
  navigation?: any;
}

const Header = (props: headerInterface) => {
    const {t} = useTranslation();
    const [modalShow, setModalShow] = useState(false);
    const [showSetting, setShowSetting] = useState(false);
    const {listNotification} = useSelector((state: RootState) => ({
        listNotification: state.notification.listNotification,
    }));
    return (
        <>
            <SideMenu
                onShowSetting={() => {
                    setModalShow(false);
                    setTimeout(() => {
                        setShowSetting(true);
                    }, 500);
                }}
                onDismis={() => {
                    setModalShow(false);
                }}
                visible={modalShow}
                navigation={props.navigation}
            />
            <SideSetting
                navigation={props.navigation}
                visible={showSetting}
                onDismis={() => {
                    setShowSetting(false);
                }}
            />
            {props.typeHeader == 'normal' ? (
                <View style={{height: sizes._screenHeight * 0.11}}>
                    <Appbar.Header
                        mode={'center-aligned'}
                        style={{backgroundColor: 'transparent'}}>
                        <View style={styles.viewLogo}>
                            <Image style={styles.imgLogo} source={images.img_logo_sidoc} />
                        </View>
                        <Appbar.Content title={''} />
                        <Appbar.Action icon={icons.ic_noti} onPress={props.onPress} />
                        <Badge style={styles.mBadge} size={sizes._15sdp}>
                            {listNotification.length}
                        </Badge>
                        <TouchableOpacity
                            style={styles.avatar}
                            onPress={() => {
                                setShowSetting(true);
                            }}>
                            <Avatar.Image
                                source={{
                                    uri: AVATAR_LINK,
                                }}
                                size={Sizes._34sdp}
                            />
                        </TouchableOpacity>
                    </Appbar.Header>
                </View>
            ) : null}
            {props.typeHeader == 'search_header' ? (
                <Appbar.Header
                    mode={'center-aligned'}
                    style={{backgroundColor: colors.color_background}}>
                    <Appbar.Action
                        icon={icons.ic_menu}
                        onPress={() => {
                            setModalShow(true);
                        }}
                    />
                    <TouchableOpacity
                        style={{
                            width: sizes._screenWidth * 0.5,
                            height: sizes._40sdp,
                            flexDirection: 'row',
                            alignItems: 'center',
                            justifyContent: 'flex-start',
                            borderWidth: sizes._1sdp,
                            borderColor: colors.color_grey3,
                            borderRadius: sizes._8sdp,
                            backgroundColor: colors.color_white,
                        }}
                        onPress={props.onPressSearch}>
                        <Image
                            source={icons.ic_find}
                            style={{
                                width: sizes._24sdp,
                                height: sizes._24sdp,
                                paddingHorizontal: sizes._20sdp,
                            }}
                            resizeMode={'contain'}
                        />
                        <Text
                            style={{
                                fontSize: sizes._14sdp,
                                fontWeight: '300',
                                lineHeight: sizes._20sdp,
                                color: colors.color_grey3,
                                fontFamily: 'LexendDeca-Light',
                            }}>
                            {t('header.search')}
                        </Text>
                    </TouchableOpacity>
                    <Appbar.Content title={''} />
                    <Appbar.Action icon={icons.ic_noti} onPress={props.onPress} />
                    <Badge
                        style={{
                            position: 'absolute',
                            right: sizes._65sdp,
                            top: sizes._15sdp,
                        }}
                        size={sizes._15sdp}>
                        {listNotification.length}
                    </Badge>
                    <TouchableOpacity
                        style={{width: 32, height: 32, borderRadius: 20, marginRight: 15}}
                        onPress={() => {
                            setShowSetting(true);
                        }}>
                        <Avatar.Image
                            source={{
                                uri: AVATAR_LINK,
                            }}
                            size={32}
                        />
                    </TouchableOpacity>
                </Appbar.Header>
            ) : null}
            {props.typeHeader == 'back_header' ? (
                <Appbar.Header
                    mode={'center-aligned'}
                    style={{backgroundColor: colors.color_background}}>
                    <Appbar.BackAction
                        onPress={props.onBackPress}
                        color={colors.color_text}
                    />
                    <Appbar.Content
                        title={props.titleHeader}
                        color={colors.color_text}
                    />
                </Appbar.Header>
            ) : null}
        </>
    );
};

export default Header;
