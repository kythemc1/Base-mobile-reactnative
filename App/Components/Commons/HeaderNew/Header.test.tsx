import renderer from 'react-test-renderer';
import React from 'react';
import Header from '../Header/Header';
test('renders correctly', () => {
    const tree = renderer.create(<Header customs />).toJSON();
    expect(tree).toMatchSnapshot();
});
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: (str: string) => str,
        i18n: {
            changeLanguage: () => new Promise(() => {}),
        },
    }),

    initReactI18next: {
        type: '3rdParty',
        init: jest.fn(),
    },
}));
