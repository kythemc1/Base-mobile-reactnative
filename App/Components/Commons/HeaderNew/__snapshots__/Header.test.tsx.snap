// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#EAEAEA",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#143980",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "justifyContent": "center",
        },
        {
          "height": undefined,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "testUri": "../../../App/Assets/Images/arrow-left.png",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginRight": 115,
              "textAlign": "left",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
